diff --git a/lib/index.d.ts b/lib/index.d.ts
index 81ac71e00f20e53fa331e737372e87379247dfd1..cefda741dddf912557429c8477adc3a1cf594b57 100644
--- a/lib/index.d.ts
+++ b/lib/index.d.ts
@@ -1,6 +1,8 @@
 export default jsxA11y;
 declare namespace jsxA11y {
     export { configs };
+    export let meta: { name: string; version: string };
+    export let rules: Record<string, import("eslint").Rule.RuleModule>;
 }
 declare namespace configs {
     let recommended: {
