diff --git a/dist/client/components/layout-router.js b/dist/client/components/layout-router.js
index 9b60a453b6195a3e070fd53a0d912812cd440fd8..e769a494cfef6ff9f14a59368b3cc8b657e546c8 100644
--- a/dist/client/components/layout-router.js
+++ b/dist/client/components/layout-router.js
@@ -317,6 +317,7 @@ function HandleRedirect({ redirect  }) {
     const router = (0, _navigation).useRouter();
     (0, _react).useEffect(()=>{
         router.replace(redirect, {});
+        router.refresh();
     }, [
         redirect,
         router
diff --git a/dist/client/link.js b/dist/client/link.js
index d15ce7f4d9324d2a1d37708eda57df6186177623..2ecb95295a3f46ad51072ab1a40759f7bee198ed 100644
--- a/dist/client/link.js
+++ b/dist/client/link.js
@@ -83,6 +83,7 @@ function linkClicked(e, router, href, as, replace, shallow, scroll, locale, isAp
     if (isAppRouter) {
         // @ts-expect-error startTransition exists.
         _react.default.startTransition(navigate);
+        router.refresh();
     } else {
         navigate();
     }
diff --git a/dist/server/web/spec-extension/response.d.ts b/dist/server/web/spec-extension/response.d.ts
index 268f52b1f12c2f1bb9d71f7c51d8dd6cf3194ac8..2b6b94bd8b68338be31b89cdbee01d7e8dbee9b4 100644
--- a/dist/server/web/spec-extension/response.d.ts
+++ b/dist/server/web/spec-extension/response.d.ts
@@ -2,14 +2,15 @@ import type { I18NConfig } from '../../config-shared';
 import { NextURL } from '../next-url';
 import { ResponseCookies } from './cookies';
 declare const INTERNALS: unique symbol;
-export declare class NextResponse extends Response {
+export declare class NextResponse<B = void> extends Response {
     [INTERNALS]: {
         cookies: ResponseCookies;
         url?: NextURL;
+        B: B;
     };
     constructor(body?: BodyInit | null, init?: ResponseInit);
     get cookies(): ResponseCookies;
-    static json(body: any, init?: ResponseInit): NextResponse;
+    static json<T>(body: T, init?: ResponseInit): NextResponse<T>;
     static redirect(url: string | NextURL | URL, init?: number | ResponseInit): NextResponse;
     static rewrite(destination: string | NextURL | URL, init?: MiddlewareResponseInit): NextResponse;
     static next(init?: MiddlewareResponseInit): NextResponse;