diff --git a/dist/server/web/spec-extension/response.d.ts b/dist/server/web/spec-extension/response.d.ts
index 268f52b1f12c2f1bb9d71f7c51d8dd6cf3194ac8..2b6b94bd8b68338be31b89cdbee01d7e8dbee9b4 100644
--- a/dist/server/web/spec-extension/response.d.ts
+++ b/dist/server/web/spec-extension/response.d.ts
@@ -2,14 +2,15 @@ import type { I18NConfig } from '../../config-shared';
 import { NextURL } from '../next-url';
 import { ResponseCookies } from './cookies';
 declare const INTERNALS: unique symbol;
-export declare class NextResponse extends Response {
+export declare class NextResponse<B = void> extends Response {
     [INTERNALS]: {
         cookies: ResponseCookies;
         url?: NextURL;
+        B: B;
     };
     constructor(body?: BodyInit | null, init?: ResponseInit);
     get cookies(): ResponseCookies;
-    static json(body: any, init?: ResponseInit): NextResponse;
+    static json<T>(body: T, init?: ResponseInit): NextResponse<T>;
     static redirect(url: string | NextURL | URL, init?: number | ResponseInit): NextResponse;
     static rewrite(destination: string | NextURL | URL, init?: MiddlewareResponseInit): NextResponse;
     static next(init?: MiddlewareResponseInit): NextResponse;